geode-assembly/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/StatusLocatorCommandDunitTest.java

```diff
@@ -0,0 +1,242 @@
+
+    LocatorState state = locator.waitOnStatusResponse(TIMEOUT, INTERVAL, SECONDS);
+
+    CommandResult result = gfsh.executeCommand(STATUS_LOCATOR + " --name=" + locator.getMemberId());
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.OK);
+    assertStatusCommandOutput(result.getMessageFromContent(), state);
+  }
+
+  @Test
+  public void testWithInvalidMemberID() throws Exception {
+    final String expectedError =
+        CliStrings.format(CliStrings.STATUS_LOCATOR__NO_LOCATOR_FOUND_FOR_MEMBER_ERROR_MESSAGE,
+            locator.getMemberId() + "1");
+
+    gfsh.connectAndVerify(locator.getPort(), PortType.locator);
+
+    CommandResult result =
+        gfsh.executeCommand(STATUS_LOCATOR + " --name=" + locator.getMemberId() + "1");
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.ERROR);
+    assertThat(result.getMessageFromContent()).contains(expectedError);
+  }
+
+  @Test
+  public void testWithDirOnline() throws Exception {
+    gfsh.connectAndVerify(locator.getPort(), PortType.locator);
+
+    LocatorState state = locator.waitOnStatusResponse(TIMEOUT, INTERVAL, SECONDS);
+
+    CommandResult result =
+        gfsh.executeCommand(STATUS_LOCATOR + " --dir=" + locator.getWorkingDirectory());
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.OK);
+    assertStatusCommandOutput(result.getMessageFromContent(), state);
+  }
+
+  @Test
+  public void testWithDirOffline() throws Exception {
+    if (gfsh.isConnected()) {
+      gfsh.disconnect();
+    }
+
+    LocatorState state = locator.waitOnStatusResponse(TIMEOUT, INTERVAL, SECONDS);
+
+    CommandResult result =
+        gfsh.executeCommand(STATUS_LOCATOR + " --dir=" + locator.getWorkingDirectory());
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.OK);
+    assertStatusCommandOutput(result.getMessageFromContent(), state);
+  }
+
+  @Test
+  public void testWithInvalidDirOnline() throws Exception {
+    final String expectedError =
+        Launcher_Builder_WORKING_DIRECTORY_NOT_FOUND_ERROR_MESSAGE.toString(LOCATOR_TERM_NAME);
+
+    gfsh.connectAndVerify(locator.getPort(), PortType.locator);
+
+    CommandResult result = gfsh.executeCommand(STATUS_LOCATOR + " --dir=/invalid/working/dir");
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.ERROR);
+    assertThat(result.getMessageFromContent()).contains(expectedError);
+  }
+
+  @Test
+  public void testWithInvalidDirOffline() throws Exception {
+    final String expectedError =
+        Launcher_Builder_WORKING_DIRECTORY_NOT_FOUND_ERROR_MESSAGE.toString(LOCATOR_TERM_NAME);
+
+    if (gfsh.isConnected()) {
+      gfsh.disconnect();
+    }
+
+    CommandResult result = gfsh.executeCommand(STATUS_LOCATOR + " --dir=/invalid/working/dir");
+
+    assertThat(result.getStatus()).isEqualTo(Result.Status.ERROR);
+    assertThat(result.getMessageFromContent()).contains(expectedError);
+  }
+
+  @AfterClass
+  public static void after() {
+    locator.stop();
+  }
```

Comment:
I prefer to see all the `@BeforeClass`, `@AfterClass`, `@Before`, `@After`,  and `@Rule` methods grouped at the top of the file so you can see all the setup and teardown without scanning back and forth through the file.

Reply:
✅ it has been moved